### ISSUE TITLE
$pid Crash bug on bash,on dash

### ISSUE THREAD

**Issue by chuanjiao10** [PowerShell/PowerShell#10410]
This bug seems to be only related to linux.

Steps to reproduce
------------------
output $pid in command line
pwsh -c "$pid"
pwsh -c "echo $pid"

The command returns normally in the following situations
------------------
powershell v5.1 ,,,, v6.2.2 in win10 ,normal. 

in pwsh ,run 【pwsh -c "$pid";pwsh -c "echo $pid"】，normal 

pwsh -c ‘$pid‘
pwsh -c ’echo $pid’

Expected behavior
-----------------
return a number

Actual behavior
---------------

--------------------------------------------------------------------------------

#pwsh 6.05 6.1.5 6.2.2 preview 7.0.3

```
pwsh -c "echo $pid"

cmdlet Write-Output at command pipeline position 1
Supply values for the following parameters:
InputObject: 
```
--------------------------------------------------------------------------------

 

# pwsh 6.05 on centos7
pwsh -c "$pid"
#return nothing.This is different from other versions.

# pwsh 6.1.5 on centos7 
# pwsh 6.2.2 on centos7,on ubuntu1604 
```
root@ubuntu1604:~# pwsh -c "$pid"

Unhandled Exception: System.ArgumentNullException: Value cannot be null.
Parameter name: value
   at System.String.IndexOf(String value, Int32 startIndex, Int32 count, StringComparison comparisonType)
   at Microsoft.PowerShell.CommandLineParameterParser.MatchSwitch(String switchKey, String match, String smallestUnambiguousMatch)
   at Microsoft.PowerShell.CommandLineParameterParser.EarlyParseHelper(String[] args)
   at Microsoft.PowerShell.CommandLineParameterParser.EarlyParse(String[] args)
   at Microsoft.PowerShell.UnmanagedPSEntry.Start(String consoleFilePath, String[] args, Int32 argc)
   at Microsoft.PowerShell.ManagedPSEntry.Main(String[] args) in /PowerShell/src/powershell/Program.cs:line 22
Aborted (core dumped)
```

# pwsh preview 7.0.3 on centos7 
```
[root@ps62 ~]# pwsh-preview -c "$pid"
Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'value')
   at System.String.IndexOf(String value, Int32 startIndex, Int32 count, StringComparison comparisonType)
   at System.String.IndexOf(String value, StringComparison comparisonType)
   at Microsoft.PowerShell.CommandLineParameterParser.MatchSwitch(String switchKey, String match, String smallestUnambiguousMatch)
   at Microsoft.PowerShell.CommandLineParameterParser.EarlyParse(String[] args)
   at Microsoft.PowerShell.UnmanagedPSEntry.Start(String consoleFilePath, String[] args, Int32 argc)
   at Microsoft.PowerShell.ManagedPSEntry.Main(String[] args)
已放弃
```

Environment data
----------------
centos76 + ps v6.0.5 + console #has this bug

centos76 + ps v6.1.5 + console #has this bug

centos76 + ps v6.2.2 + console #has this bug

ubuntu1604 + ps v6.2.2 + console #has this bug

centos76 + ps v7.0 preview3 + console #has this bug

win10 + ps v6.2.2 #no this bug

win10 + powershell 5.1 #no this bug

 
reporter
----------------
By the chinese powershell study & research group. QQ group No. is 183173532

**Comment by chuanjiao10** [comment]
same problem in pwsh 7.04 
```
[root@ps62 ~]# pwsh-preview -c "$pid"
Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'value')
   at System.String.IndexOf(String value, Int32 startIndex, Int32 count, StringComparison comparisonType)
   at System.String.IndexOf(String value, StringComparison comparisonType)
   at Microsoft.PowerShell.CommandLineParameterParser.MatchSwitch(String switchKey, String match, String smallestUnambiguousMatch)
   at Microsoft.PowerShell.CommandLineParameterParser.EarlyParse(String[] args)
   at Microsoft.PowerShell.UnmanagedPSEntry.Start(String consoleFilePath, String[] args, Int32 argc)
   at Microsoft.PowerShell.ManagedPSEntry.Main(String[] args)
已放弃
[root@ps62 ~]# pwsh-preview
PowerShell 7.0.0-preview.4
Copyright (c) Microsoft Corporation. All rights reserved.

https://aka.ms/powershell
Type 'help' to get help.

PS /root> exit
[root@ps62 ~]# 

```

**Comment by iSazonov** [comment]
@mklement0 I wonder is it argument parse issue?

**Comment by mklement0** [comment]
@iSazonov:

This is not a PowerShell problem.

The issue is that POSIX-like shells _also_ expand `$`-prefixed tokens inside `"..."`, so a shell such as `bash` expands `$pid` _before_ PowerShell sees it; since `$pid` is (typically) not defined (as an aside: the equivalent would be `$$`), it expands to the empty string, so the literal command string that PowerShell sees is the _empty string_ with argument `"$pid"` and `echo ` with argument `"echo $pid"` - hence the symptoms.

In short: you need to ensure that the _calling shell_ doesn't interpret your command string up front, which in POSIX-like shells can be done in of two ways:

* `\`-escaping of `$` chars: 
  * `pwsh -c "echo \$pid"`

* more simply, in this case, since no interpolation is needed: single-quoting: 
  * `pwsh -c 'echo $pid`.

That said, the crash and unhelpful error message you currently get with `pwsh -c ''` should obviously be fixed - see #10684

**Comment by iSazonov** [comment]
@mklement0 Thanks! I tried this from PowerShell session and was wonder that it worked. :-)
